util/findLostPatches.py

```diff
@@ -0,0 +1,165 @@
+parser.add_option("-i", "--jbide", dest="jbidefixversion", help="JBIDE Fix Version, eg., 4.1.0.qualifier")
+parser.add_option("-d", "--jbds", dest="jbdsfixversion", help="JBDS Fix Version, eg., 7.0.0.qualifier")
+parser.add_option("-t", "--task", dest="taskdescription", help="Task Summary, eg., \"Code Freeze + Branch\"")
+parser.add_option("-f", "--taskfull", dest="taskdescriptionfull", help="Task Description, eg., \"Please perform the following tasks...\"");
+# see createTaskJIRAs.py.examples.txt for examples of taskdescriptionfull
+
+(options, args) = parser.parse_args()
+
+if not options.username or not options.password or not options.jiraserver or not options.jbidefixversion or not options.jbdsfixversion or not options.taskdescription:
+    parser.error("Must to specify ALL commandline flags")
+    
+jiraserver = options.jiraserver
+jira = JIRA(options={'server':jiraserver}, basic_auth=(options.username, options.password))
+
+jbide_fixversion = options.jbidefixversion
+jbds_fixversion = options.jbdsfixversion
+taskdescription = options.taskdescription
+taskdescriptionfull = options.taskdescriptionfull
+if not options.taskdescriptionfull:
+    taskdescriptionfull = options.taskdescription
+
+## The jql query across for all task issues
+tasksearchquery = '((project in (JBDS) and fixVersion = "' + jbds_fixversion + '") or (project in (JBIDE) and fixVersion = "' + jbide_fixversion + '")) AND labels = task'
+
+tasksearch = jiraserver + '/issues/?jql=' + urllib.quote_plus(tasksearchquery)
+
+rootJBDS_dict = {
+    'project' : { 'key': 'JBDS' },
+    'summary' :     'For JBDS ' + jbds_fixversion + ': ' + taskdescription,
+    'description' : 'For JBDS ' + jbds_fixversion + ': ' + taskdescriptionfull + '\n\n[Search for all task JIRA|' + tasksearch + ']',
+    'issuetype' : { 'name' : 'Task' },
+    'priority' : { 'name' :'Blocker'},
+    'fixVersions' : [{ "name" : jbds_fixversion }],
+    'components' : [{ "name" : "installer" }],
+    'labels' : [ "task" ],
+    }
+rootJBDS = jira.create_issue(fields=rootJBDS_dict)
+print("Task JIRA created for this milestone include:")
+print("")
+
+print("JBDS              : " + jiraserver + '/browse/' + rootJBDS.key)
+
+rootJBIDE_dict = {
+    'project' : { 'key': 'JBIDE' },
+    'summary' :     'For JBIDE ' + jbide_fixversion + ': ' + taskdescription,
+    'description' : 'For JBIDE ' + jbide_fixversion + ': ' + taskdescriptionfull + '\n\n[Search for all task JIRA|' + tasksearch + ']\n\nSee also: ' + rootJBDS.key,
+    'issuetype' : { 'name' : 'Task' },
+    'priority' : { 'name' :'Blocker'},
+    'fixVersions' : [{ "name" : jbide_fixversion }],
+    'components' : [{ "name" : "build" }],
+    'labels' : [ "task" ]
+    }
+rootJBIDE = jira.create_issue(fields=rootJBIDE_dict)
+print("JBoss Tools       : " + jiraserver + '/browse/' + rootJBIDE.key)
+
+## map from descriptive name to list of JBIDE and/or JBDS components.
+JBT_components = {
+
+# active projects
```

Comment:
We should figure out how to externalize this to another file so that this script and the ones in jbdevstudio-ci/bin/*.py can reuse the mappings w/o having to update in multiple places.